src/ops/kernels/conv_2d_kernels.cu

```diff
@@ -0,0 +1,477 @@
+                            float const *filter_ptr,
+                            float const *bias_ptr) {
+  // printf("fwdAlgo(%d), bwdFilterALgo(%d), bwdDataAlgo(%d)\n",
+  // (int)m->fwdAlgo,(int) m->bwdFilterAlgo,(int) m->bwdDataAlgo);
+  cudaStream_t stream;
+  checkCUDA(get_legion_stream(&stream));
+
+  cudaEvent_t t_start, t_end;
+  if (m->profiling) {
+    cudaEventCreate(&t_start);
+    cudaEventCreate(&t_end);
+    cudaEventRecord(t_start, stream);
+  }
+
+  Internal::forward_kernel(
+      m, input_ptr, output_ptr, filter_ptr, bias_ptr, stream);
+  if (m->profiling) {
+    cudaEventRecord(t_end, stream);
+    checkCUDA(cudaEventSynchronize(t_end));
+    print_tensor<float>(input_ptr, 16, "[Conv2D:forward:input]");
+    print_tensor<float>(filter_ptr, 16, "[Conv2D:forward:kernel]");
+    print_tensor<float>(bias_ptr, 16, "[Conv2D:forward:bias]");
+    print_tensor<float>(output_ptr, 16, "[Conv2D:forward:output]");
+    float elapsed = 0;
+    checkCUDA(cudaEventElapsedTime(&elapsed, t_start, t_end));
+    cudaEventDestroy(t_start);
+    cudaEventDestroy(t_end);
+    printf("%s [Conv2D] forward time (CF) = %.2fms\n", m->op_name, elapsed);
+  }
+}
+
+void backward_kernel_wrapper(Conv2DMeta const *m,
+                             float const *input_ptr,
+                             float *input_grad_ptr,
+                             float const *output_ptr,
+                             float *output_grad_ptr,
+                             float const *kernel_ptr,
+                             float *kernel_grad_ptr,
+                             float *bias_grad_ptr) {
+  cudaStream_t stream;
+  checkCUDA(get_legion_stream(&stream));
+
+  cudaEvent_t t_start, t_end;
+  if (m->profiling) {
+    cudaEventCreate(&t_start);
+    cudaEventCreate(&t_end);
+    cudaEventRecord(t_start, stream);
+  }
+
+  Internal::backward_kernel(m,
+                            input_ptr,
+                            input_grad_ptr,
+                            output_ptr,
+                            output_grad_ptr,
+                            kernel_ptr,
+                            kernel_grad_ptr,
+                            bias_grad_ptr,
+                            stream);
+  if (m->profiling) {
+    cudaEventRecord(t_end, stream);
+    checkCUDA(cudaEventSynchronize(t_end));
+    float elapsed = 0;
+    checkCUDA(cudaEventElapsedTime(&elapsed, t_start, t_end));
+    cudaEventDestroy(t_start);
+    cudaEventDestroy(t_end);
+    printf("%s [Conv2D] backward time = %.2fms\n", m->op_name, elapsed);
+    // print_tensor<4, float>(acc_output_grad.ptr, acc_output_grad.rect,
+    // "[Conv2D:backward:output_grad]"); print_tensor<4,
+    // float>(acc_kernel_grad.ptr, acc_kernel_grad.rect,
+    // "[Conv2D:backward:kernel_grad]"); print_tensor<1,
+    // float>(acc_bias_grad.ptr, acc_bias_grad.rect,
+    // "[Conv2D:backward:bias_grad]"); print_tensor<4,
+    // float>(acc_input_grad.ptr, acc_input_grad.rect,
+    // "[Conv2D:backward:input_grad]");
```

Comment:
Remove?

Reply:
> The .cpp file seem to mimic its corresponding .cu file. Is it auto-generated? If not, should that also be modified for this PR?

The `.cpp` files are for running FlexFlow on AMD GPUs. They should be updated as well. 

Is this PR ready to be merged? I am about to push some changes to the fused.cc and fused.cu, which may conflict with some of your changes.

Reply:
Not yet, go ahead and push those changes and I'll deal with any conflicts